### PR TITLE
feat: Add readme and local run bash script for E2E tests

### PR DESCRIPTION
**Pull request by catch-21**
### Description

Add readme to E2E directory to explain the purpose, how to run locally, and how they are run in the CI pipeline.
Add a bash script for running E2E tests locally.

### Linked Issues/Tasks

Add any links to GitHub issues or Asana tasks that are relevant to this pull request.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [x] Detox test
- [ ] Unit test
- [ ] No test

### Screenshot / Video

None

### QA Notes

None

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- For iOS Tests:
+  - The host must be running macOS.
+  - iOS Simulator must be installed, which comes with Xcode.
+- For Android Tests:
+  - An Android Virtual Device (AVD) must be set up.
+  - The Android Debug Bridge (ADB) must be installed.
+
+Additionally, you may need to update the `devices.emulator.device.avdName` and/or `devices.simulator.device.type` settings in the .detoxrc.js configuration file to match your environment.
+
+### Using `act`
```
Comment (limpbrains) [inline e2e/README.md]: Did you try act? Does it work with our e2e tests?
I would rather suggest your bash script by default, looks solid

**Review by limpbrains — APPROVED** [review]

File: e2e/run.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+if [ "$BUILD_OPTION" == "build" ]; then
+  echo "Building the app..."
+  yarn e2e:build:$PLATFORM-$BUILD_TYPE
+else
+  echo "Skipping build..."
+fi
+
+# Start the Docker Compose environment
+docker compose -f docker/docker-compose.yml up -d
+sleep 2 # short wait for Electrum and LND
```
Comment (limpbrains) [inline e2e/run.sh]: This will fail if the user is running tests for the first time, as the images have not been downloaded yet.

Comment (catch-21) [reply e2e/run.sh]: If it doesn't have the images then it will pull them before proceeding to start containers. So I think it is ok?

Comment (catch-21) [inline e2e/run.sh]: I could improve the wait here. I didn't need one at all on my machine as all services are ready before the first test begins.

Comment (catch-21) [reply e2e/run.sh]: addressed

File: e2e/README.md (hunk 1 above)
Comment (catch-21) [reply e2e/README.md]: Yes, it works flawlessly, for me at least. I've swapped the order of the two so bash script is before act.

**Comment by JeanlChristophe** [comment]
@catch-21 please do no forget to add Projects information and sprint

**Pull request merged**